components/automate-ui/src/app/pages/automate-settings/automate-settings.component.spec.ts

```diff
@@ -88,78 +145,121 @@ describe('AutomateSettingsComponent', () => {
-        every: '60m'
-      };
-      mockJobSchedulerStatus = new JobSchedulerStatus(true, [
-        jobMissingNodes,
-        jobMissingNodesForDeletion
-      ]);
-    });
 
-    it('updates the "missingNodes" form group correctly', () => {
-      component.updateForm(mockJobSchedulerStatus);
-
-      const missingNodesValues = component.automateSettingsForm
-        .controls.missingNodes.value;
+      const infraNestedForms: IngestJob = {
+        name: 'periodic_purge_timeseries',
+        category: 'infra',
+        disabled: false,
+        threshold: '',
+        purge_policies: {
+          elasticsearch: [
+            {
+              name: 'actions',
+              older_than_days: 22, // default is 30, since disabled
+              disabled: true       // is true older than should be null
+            },
+            {
+              name: 'converge-history',
+              older_than_days: 12,
+              disabled: false
+            }
+          ]
+        }
+      };
 
-      expect(missingNodesValues.disable).toEqual(true);
-      expect(missingNodesValues.threshold).toEqual('60');
-      expect(missingNodesValues.unit).toEqual('m');
-    });
+      const complianceForms: IngestJob = {
+        category: 'compliance',
+        name: 'periodic_purge',
+        threshold: '',
+        disabled: true,
+        purge_policies: {
+          elasticsearch: [
+            {
+              name: 'compliance-reports',
+              older_than_days: 105,
+              disabled: false
+            },
+            {
+              name: 'compliance-scans',
+              older_than_days: 92,
+              disabled: false
+            }
+          ]
+        }
+      };
 
-    it('updates the "deleteMissingNodes" form group correctly', () => {
-      component.updateForm(mockJobSchedulerStatus);
+      const clientRunsRemoveData: IngestJob = {
+        category: 'infra',
+        name: 'missing_nodes',
+        disabled : false,
+        threshold : '7d'
+      };
 
-      const deleteMssingNodesValues = component.automateSettingsForm
-        .controls.deleteMissingNodes.value;
+      const clientRunsLabelMissing: IngestJob = {
+        category: 'infra',
+        name: 'missing_nodes_for_deletion',
+        disabled: false,
+        threshold: '14m'
+      };
 
-      expect(deleteMssingNodesValues.disable).toEqual(false);
-      expect(deleteMssingNodesValues.threshold).toEqual('24');
-      expect(deleteMssingNodesValues.unit).toEqual('h');
+      mockJobSchedulerStatus = new JobSchedulerStatus([
+        eventFeedRemoveData,
+        infraNestedForms,
+        clientRunsRemoveData,
+        clientRunsLabelMissing,
+        complianceForms
+      ]);
     });
 
-    it('does not updates the "eventFeed" form group', () => {
-      component.updateForm(mockJobSchedulerStatus);
+    using([
+      ['eventFeedRemoveData', false, 53 ],
+      ['eventFeedServerActions', true, undefined ], // Infra purge_timeseries -> server_actions
```

Comment:
I think this test can be improved from what you've mentioned here, and I'll attempt to make some improvements, but I waned to provide some clarity first.

`eventFeedServerActions` is not a special case, I was using it to test out a different scenario, i.e. the form is disabled.  What I should really be doing is testing all of them in both fashions.

When a formControl is disabled in the UI, Angular's formGroup actually removes that formControl from the formGroup, so the value comes through as undefined instead of as a value, which is what we expect.  When a group is disabled, we don't actually need that value for the time being.